ddtrace/contrib/internal/ray/span_manager.py

```diff
@@ -0,0 +1,220 @@
+from contextlib import contextmanager
+from itertools import chain
+import threading
+import time
+
+from ddtrace import config
+from ddtrace import tracer
+from ddtrace._trace.span import Span
+from ddtrace.constants import SPAN_KIND
+from ddtrace.ext import SpanKind
+
+from .utils import _inject_ray_span_tags
+
+
+@contextmanager
+def long_running_ray_span(span_name, service, span_type, child_of=None, activate=True):
+    """Context manager that handles Ray span creation and long-running span lifecycle"""
+    span = tracer.start_span(name=span_name, service=service, span_type=span_type, child_of=child_of, activate=activate)
+    span.set_tag_str(SPAN_KIND, SpanKind.CONSUMER)
+    _inject_ray_span_tags(span)
+    start_long_running_span(span)
+
+    try:
+        yield span
+    except Exception as e:
+        span.set_exc_info(type(e), e, e.__traceback__)
+        raise
+    finally:
+        stop_long_running_span(span)
+
+
+class LongRunningJobManager:
+    def __init__(self):
+        self._timers = {}
+        # {submission_id: {(trace_id, span_id): Span}}
+        self._job_spans = {}
+        # {submission_id: Span}
+        self._root_spans = {}
+        self._lock = threading.Lock()
+
+    def _get_submission_id(self, span):
+        submission_id = span.get_tag("ray.submission_id")
+        return submission_id if submission_id else None
+
+    def _emit_partial_span(self, span):
+        partial_span = self._recreate_job_span(span)
+        partial_span.set_metric("_dd.partial_version", time.time_ns())
+        partial_span.finish()
+
+        # Sending spans which are waiting for long running spans to finish
+        aggregator = tracer._span_aggregator
+        finished_children = []
```

Comment:
```suggestion
        finished_spans = []
```
I am sorry to keep nitpicking on names, but I think `finished_spans` is a better name than `finished_children`, since these spans are not necessarily descendants of the long-running span that we are emitting a partial version of.

Reply:
addressed in [a2274f2](https://github.com/DataDog/dd-trace-py/pull/14511/commits/a2274f28673c79c24fb31f2f6c1b5bf78e0eb6da)